### PR TITLE
R stable enable virtio video

### PR DESCRIPTION
**Pull request by Shao-Feng**

### CODE REVIEW

**Comment by Shao-Feng** [comment]
Hi @ceiba1985 Please help to review

File: codecs/v4l2-codec2/media_c2_v4l2_hal.te [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+allow media_c2_v4l2_hal hwservicemanager_prop:file { open read getattr map };
+
+allow media_c2_v4l2_hal hwservicemanager:binder call;
+allow media_c2_v4l2_hal tombstoned_crash_socket:sock_file write;
+allow media_c2_v4l2_hal hal_codec2_hwservice:hwservice_manager { find add };
+
+allow media_c2_v4l2_hal hidl_base_hwservice:hwservice_manager add;
+
+allow media_c2_v4l2_hal mediaserver:binder transfer;
```
Comment (ceiba1985) [inline codecs/v4l2-codec2/media_c2_v4l2_hal.te]: replace line 18 and l9 with: binder_call(mediaserver, media_c2_v4l2_hal)

File: codecs/v4l2-codec2/media_c2_v4l2_hal.te [hunk 2]
```diff
@@ -0,0 +1,22 @@
+type media_c2_v4l2_hal, domain;
+
+type media_c2_v4l2_hal_exec, exec_type, file_type, vendor_file_type;
+init_daemon_domain(media_c2_v4l2_hal)
+
+vndbinder_use(media_c2_v4l2_hal)
+hal_client_domain(media_c2_v4l2_hal, hal_codec2)
+hal_server_domain(media_c2_v4l2_hal, hal_codec2)
+
+allow media_c2_v4l2_hal hwservicemanager_prop:file { open read getattr map };
```
Comment (ceiba1985) [inline codecs/v4l2-codec2/media_c2_v4l2_hal.te]: use: get_prop(media_c2_v4l2_hal, hwservicemanager_prop)

File: codecs/v4l2-codec2/file_contexts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+/vendor/bin/hw/android\.hardware\.media\.c2@1\.0-service-v4l2   u:object_r:media_c2_v4l2_hal_exec:s0
+/dev/video[0-9] u:object_r:video_device:s0
```
Comment (ceiba1985) [inline codecs/v4l2-codec2/file_contexts:2]: I thought video_device is already defined in aosp.

File: codecs/v4l2-codec2/media_c2_v4l2_hal.te [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+allow media_c2_v4l2_hal hwservicemanager:binder call;
+allow media_c2_v4l2_hal tombstoned_crash_socket:sock_file write;
+allow media_c2_v4l2_hal hal_codec2_hwservice:hwservice_manager { find add };
+
+allow media_c2_v4l2_hal hidl_base_hwservice:hwservice_manager add;
+
+allow media_c2_v4l2_hal mediaserver:binder transfer;
+allow mediaserver media_c2_v4l2_hal:binder call;
+
+allow media_c2_v4l2_hal cgroup_v2:dir search;
```
Comment (ceiba1985) [inline codecs/v4l2-codec2/media_c2_v4l2_hal.te]: Where cgoup_v2 is defined?

Comment (dyang23) [reply codecs/v4l2-codec2/media_c2_v4l2_hal.te]: > Where cgoup_v2 is defined?

http://inlubt0232.iind.intel.com:8080/source/xref/CELADON/system/sepolicy/prebuilts/api/31.0/private/compat/30.0/30.0.ignore.cil?r=db5e6c24#26

**Comment by harvan** [comment]
we should add sepolicy for v4l2.encoder:
allow media_c2_v4l2_hal hal_graphics_mapper_hwservice:hwservice_manager find;
hal_client_domain(media_c2_v4l2_hal, hal_graphics_allocator)

allow media_c2_v4l2_hal hal_graphics_allocator_default_tmpfs:file { read write map };
allow media_c2_v4l2_hal gpu_device:dir search;
allow media_c2_v4l2_hal gpu_device:chr_file { open read write ioctl getattr map};

allow media_c2_v4l2_hal tombstone_data_file:file append;
allow media_c2_v4l2_hal tombstoned:fd use;
allow media_c2_v4l2_hal tombstoned:unix_stream_socket connectto;

File: codecs/v4l2-codec2/media_c2_v4l2_hal.te (hunk 4 above)
Comment (ceiba1985) [reply codecs/v4l2-codec2/media_c2_v4l2_hal.te]: I think this patch is for R, but the link you posted out is S.

**Comment by Shao-Feng** [comment]
Updated， Please review again

**Comment by Shao-Feng** [comment]
Updated and verified in R_stable. Please help to review.

**Review by dyang23 — APPROVED** [review]

**Review by ceiba1985 — APPROVED** [review]

**Review by dyang23 — APPROVED** [review]
LGTM

File: codecs/v4l2-codec2/media_c2_v4l2_hal.te [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+allow media_c2_v4l2_hal gpu_device:chr_file { open read write ioctl getattr map};
+
+allow media_c2_v4l2_hal tombstone_data_file:file append;
+allow media_c2_v4l2_hal tombstoned:fd use;
+allow media_c2_v4l2_hal tombstoned:unix_stream_socket connectto;
+
+allow media_c2_v4l2_hal dumpstate:fd use;
+
+#============= mediaswcodec ==============
+allow mediaswcodec gpu_device:chr_file map;
```
Comment (ceiba1985) [inline codecs/v4l2-codec2/media_c2_v4l2_hal.te]: Please move this rule to mediaswcodec.te for better management.

File: codecs/v4l2-codec2/media_c2_v4l2_hal.te [hunk 6]
```diff
@@ -7,16 +7,35 @@ vndbinder_use(media_c2_v4l2_hal)
...
 allow media_c2_v4l2_hal video_device:chr_file { open read write ioctl };
+
+allow media_c2_v4l2_hal hal_graphics_mapper_hwservice:hwservice_manager find;
+hal_client_domain(media_c2_v4l2_hal, hal_graphics_allocator)
+
+allow media_c2_v4l2_hal hal_graphics_allocator_default_tmpfs:file { read write map };
+allow media_c2_v4l2_hal gpu_device:dir search;
+allow media_c2_v4l2_hal gpu_device:chr_file { open read write ioctl getattr map};
+
+allow media_c2_v4l2_hal tombstone_data_file:file append;
```
Comment (ceiba1985) [inline codecs/v4l2-codec2/media_c2_v4l2_hal.te]: please replace lines 34 to lines 38 with crash_dump_fallback(media_c2_v4l2_hal)

**Comment by Shao-Feng** [comment]
@ceiba1985 Updated, please help to review.

**Comment by Shao-Feng** [comment]
Hi @lchen43 @JeevakaPrabu @kaushlen @sgnanase Please help to review

**Review by ceiba1985 — APPROVED** [review]

**Review by lchen43 — APPROVED** [review]

**Pull request merged**